### DISCUSSION TITLE
Matplotlib dependency

### DISCUSSION THREAD

**Discussion by FranzForstmayr** [scikit-rf/scikit-rf#778] [General]
Since #628 matplotlib is a required dependency (again).

Should we require matplotlib for all installations, or should we treat it as optional dependency?
I have at least some applications which don't use matplotlib but plotly as backend, so I would not need matplotlib.

Open for discussion.

**Comment by jhillairet** [comment]
Indeed, I think it should not be a strict requirement. I believe this dependency could be made optional in `deembbed.py` as well.

**Reply by mhuser** [reply]
Ooops is it because of the IEEEP370 methods ? Indeed they clone matlab methods that can output plots in some case to show internal process data (kind of verbose mode). Can I do that without making matplotlib a dependance ?

**Reply by FranzForstmayr** [reply]
Yes, you can do it like in vectorfitting.py with the Check plotting decorator.
One could move the decorator in a common place and Import from both files.

**Reply by mhuser** [reply]
Sounds optimal!

**Reply by mhuser** [reply]
OK I am giving it a try here: #782

**Comment by jhillairet** [comment]
Solved in #785 by @FranzForstmayr
